.github/scripts/test.sh

```diff
@@ -0,0 +1,23 @@
+# Actions invokes script with: bash -e <script>
+
```

Comment:
Rather than running `make package` to run the code-quality tests, run `make test` in a development shell, using a technique similar to what we used in `format-check.sh`.

Here, the development shell gets re-used for each Python release to test, but with the correct Python interpreter version installed for each test, and the `uw` entry-point script updated to point to the new Python interpreter. 

(Unfortunately, `setuptools` hardcodes the full path to the Python interpreter in the entry-point scripts, e.g. `#!/home/maddenp/sw/conda/envs/DEV-uwtools/bin/python3.11`, so that has to be updated via the `pip` command seen below when the Python version changes.)

[Here is a sample CI run of this script.](https://github.com/ufs-community/workflow-tools/actions/runs/6815371576/job/18534320286#step:4:1)